File: Hospital Information System_ui/src/app/Models/PagedResponse.ts
import { from } from "rxjs";
import { Response } from 'src/app/Models/Response'


export class PagedResponse<T> extends Response<T> {


  pageNumber: number;
  pageSize: number;
  totalPages: number;
  totalRecords: number;

  
  constructor(data: T,
    succeeded: boolean,
    errors: string[],
    message: string,
    pageNumber: number,
    pageSize: number,
    totalPages: number,
    totalRecords: number) {


    super(data, succeeded, errors, message);


    this.pageNumber = pageNumber;
    this.pageSize = pageSize;
    this.totalPages = totalPages;
    this.totalRecords = totalRecords;

  }
}

File: Hospital Information System API/HospitalInformationSystemAPI/Wrappers/PagedResponse.cs
using HospitalInformationSystemAPI.Filters;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace HospitalInformationSystemAPI.Wrappers
{
    public class PagedResponse<T> : Response<T>
    {
        public int PageNumber { get; set; }
        public int PageSize { get; set; }

        public int TotalPages { get; set; }
        public int TotalRecords { get; set; }

        public PagedResponse(T data, PaginationFilter filter, int totalRecords) : base(data)
        {
            this.PageNumber = filter.PageNumber;
            this.PageSize = filter.PageSize;

            this.TotalRecords = totalRecords;
            this.TotalPages = totalRecords / this.PageSize + (totalRecords % this.PageSize > 0 ? 1 : 0);
        }
    }
}
